packages/php/blueprint/src/ResourceStorages/OrgPluginResourceStorage.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint\ResourceStorages;
+
+class OrgPluginResourceStorage implements ResourceStorage {
+	public function download( $slug ): string {
+		return $this->download_url( $this->get_download_link( $slug ) );
+	}
+
+	protected function download_url( $url ) {
+		if ( ! function_exists( 'download_url' ) ) {
+			include ABSPATH . '/wp-admin/includes/file.php';
+		}
+		return \download_url( $url );
+	}
+
+	protected function get_download_link( $slug ) {
+		if ( ! function_exists( 'plugins_api' ) ) {
+			include_once ABSPATH . '/wp-admin/includes/plugin-install.php';
+		}
+		$info = \plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => $slug,
+				'fields' => array(
+					'sections' => false,
+				),
+			)
+		);
+
+		return $info->download_link;
```

Comment:
Should we check `$info` type before accessing `download_link`?  `plugins_api` might return a WP error object when it fails.

Reply:
Good question!

Updated in [Check for download_link -- it can be empty](https://github.com/woocommerce/woocommerce/pull/49763/commits/3ff19aa5fd402bf6a400e37a103e7842fc2850b7) as well